doc/aaa/AAA Improvements/AAA Improvements.md

```diff
@@ -0,0 +1,636 @@
+# AAA Improvements
+
+## Introduction
+
+This document identifies a few gaps in the current implementation of Authentication, Authorization, and Accounting (AAA) in SONiC and proposes a way to remedy them. But before we get into the crux of the matter, we will first take the time to understand when and where AAA plays a role in SONiC by examining how a typical SSH login session behaves. Understanding how a login session works is crucial to understanding why the current AAA implementation requires some changes. 
+
+### Document conventions and limitations
+
+This document is akin to a white paper with some elements of a High-Level Design document. The goal is to spark a discussion with the SONiC community regarding AAA. The main focus is on the interactions between AAA and Linux. More precisely, how PAM and NSS work together to support Authentication, Authorization, and Accounting. Advanced AAA topics such as "*command accounting*" or "*how applications can enforce authorization*" are not discussed here.
+
+In this document, the ***Management Framework*** project refers to the code located in the ***sonic-mgmt-framework*** repository.
```

Comment:
How do we protect the account creations from applications like sonic-mgmt-framework or any other applications? Do you have any suggestions?

Reply:
This is already addressed in the document. We put in place rules that prevent unauthorized applications to access the D-Bus methods. That's specified in the section titled "D-Bus and security". 

In other words, only applications member of the group "`hamd_accounts`" will be allowed to invoke the D-Bus methods to create user accounts.